streams/src/test/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilderTest.java

```diff
@@ -0,0 +1,370 @@
+    }
+
+    @Test
+    public void shouldAddTableToLatestAutoOffsetResetList() {
+        final String topicName = "topic-1";
+        final String storeName = "test-store";
+
+        builder.table(AutoOffsetReset.LATEST, null, null, null, topicName, storeName);
+
+        assertTrue(builder.internalTopologyBuilder.latestResetTopicsPattern().matcher(topicName).matches());
+        assertFalse(builder.internalTopologyBuilder.earliestResetTopicsPattern().matcher(topicName).matches());
+    }
+
+    @Test
+    public void shouldNotAddTableToOffsetResetLists() {
+        final String topicName = "topic-1";
+        final String storeName = "test-store";
+        final Serde<String> stringSerde = Serdes.String();
+
+        builder.table(null, null, stringSerde, stringSerde, topicName, storeName);
+
+        assertFalse(builder.internalTopologyBuilder.latestResetTopicsPattern().matcher(topicName).matches());
+        assertFalse(builder.internalTopologyBuilder.earliestResetTopicsPattern().matcher(topicName).matches());
+    }
+
+    @Test
+    public void shouldNotAddRegexTopicsToOffsetResetLists() {
+        final Pattern topicPattern = Pattern.compile("topic-\\d");
+        final String topic = "topic-5";
+
+        builder.stream(null, null, null, null, topicPattern);
+
+        assertFalse(builder.internalTopologyBuilder.latestResetTopicsPattern().matcher(topic).matches());
+        assertFalse(builder.internalTopologyBuilder.earliestResetTopicsPattern().matcher(topic).matches());
+
+    }
+
+    @Test
+    public void shouldAddRegexTopicToEarliestAutoOffsetResetList() {
+        final Pattern topicPattern = Pattern.compile("topic-\\d+");
+        final String topicTwo = "topic-500000";
+
+        builder.stream(AutoOffsetReset.EARLIEST, null, null, null,  topicPattern);
+
+        assertTrue(builder.internalTopologyBuilder.earliestResetTopicsPattern().matcher(topicTwo).matches());
+        assertFalse(builder.internalTopologyBuilder.latestResetTopicsPattern().matcher(topicTwo).matches());
+    }
+
+    @Test
+    public void shouldAddRegexTopicToLatestAutoOffsetResetList() {
+        final Pattern topicPattern = Pattern.compile("topic-\\d+");
+        final String topicTwo = "topic-1000000";
+
+        builder.stream(AutoOffsetReset.LATEST, null, null, null, topicPattern);
+
+        assertTrue(builder.internalTopologyBuilder.latestResetTopicsPattern().matcher(topicTwo).matches());
+        assertFalse(builder.internalTopologyBuilder.earliestResetTopicsPattern().matcher(topicTwo).matches());
+    }
+
+    @Test
+    public void kStreamTimestampExtractorShouldBeNull() throws Exception {
+        builder.stream(null, null, null, null, "topic");
+        final ProcessorTopology processorTopology = builder.internalTopologyBuilder.build(null);
+        assertNull(processorTopology.source("topic").getTimestampExtractor());
+    }
+
+    @Test
+    public void shouldAddTimestampExtractorToStreamWithKeyValSerdePerSource() throws Exception {
```

Comment:
Not sure about the name? In particular the WithKeyValueSerde - we aren't adding any serdes?